source/agora/common/EnrollmentManager.d

```diff
@@ -218,39 +247,56 @@ public class EnrollmentManager
             buffer ~= data;
         };
 
-        serializePart(this.signature_noise, dg);
+        try
+        {
+            serializePart(this.signature_noise, dg);
+        }
+        catch (Exception ex)
+        {
+            this.logMessage("Serialization error", enroll, ex);
+            return false;
+        }
 
-        () @trusted {
-            auto results = this.db.execute("SELECT EXISTS(SELECT 1 FROM node_enroll_data " ~
-                "WHERE key = ?)", "signature_noise");
-            if (results.oneValue!(bool))
-            {
-                this.db.execute("UPDATE node_enroll_data SET val = ? WHERE key = ?",
-                    buffer, "signature_noise");
-            }
-            else
-            {
-                this.db.execute("INSERT INTO node_enroll_data (key, val) VALUES (?, ?)",
-                    "signature_noise", buffer);
-            }
-        }();
+        try
+        {
+            () @trusted {
+                auto results = this.db.execute("SELECT EXISTS(SELECT 1 FROM node_enroll_data " ~
+                    "WHERE key = ?)", "signature_noise");
```

Comment:
If the key is always the same, do we really need a key to begin with ? This is a single-entry table AFAICS.

Reply:
The table can have another data such as Pre-Images. So using key makes sense.

Reply:
Right, thanks for the clarification!